relay-profiling/src/sample.rs

```diff
@@ -315,8 +318,28 @@ fn parse_profile(payload: &[u8]) -> Result<SampleProfile, ProfileError> {
     Ok(profile)
 }
 
-pub fn parse_sample_profile(payload: &[u8]) -> Result<Vec<u8>, ProfileError> {
-    let profile = parse_profile(payload)?;
+pub fn parse_sample_profile(
+    payload: &[u8],
+    tags: BTreeMap<String, String>,
+) -> Result<Vec<u8>, ProfileError> {
+    let mut profile = parse_profile(payload)?;
+
+    if let Some(transaction_name) = tags.get("transaction") {
+        if let Some(ref mut transaction) = profile.transaction {
```

Comment:
`parse_android_profile` also sets `profile.transaction_name` here, is that difference intentional?

Either way it seems like these two functions could share some code, maybe unite them and pass a boolean argument for setting `profile.transaction_name`?

Reply:
`parse_profile` in each `parse_sample_profile` and `parse_android_profile` are not returning the same type. I can't make a function which would accept both types and let me set the fields directly, can I?

Reply:
You are right, you would need a trait for that. Or refactor the `*Profile` structs to contain a common `BaseProfile` with [`#[serde(flatten)]`](https://serde.rs/attr-flatten.html#factor-out-frequently-grouped-keys). But that would be out of scope for this PR.